File: examples/java/pact_validation_with_oas/one_consumer_one_provider/provider/src/main/java/br/com/zup/pact/provider/service/AccountService.java
package br.com.zup.pact.provider.service;

import br.com.zup.pact.provider.dto.AccountDetailsDTO;
import br.com.zup.pact.provider.dto.BalanceDTO;
import java.util.List;
import java.util.Optional;

public interface AccountService {
    Optional<AccountDetailsDTO> getAccountDetailsByAccountId(Integer clientId);

    Optional<List<AccountDetailsDTO>> getAll();

    Optional<BalanceDTO> getBalanceByAccountId(Integer clientId);
}

File: example/kotlin/gRPC/provider/src/main/kotlin/br/com/zup/pact/provider/service/AccountService.kt
package br.com.zup.pact.provider.service

import br.com.zup.pact.provider.dto.AccountDetailsDTO
import br.com.zup.pact.provider.dto.BalanceDTO
import java.util.*

interface AccountService {
    fun getAll(): List<AccountDetailsDTO>
    fun findByAccountId(accountId: Int): BalanceDTO?
}
